include/tvm/build_module.h

```diff
@@ -0,0 +1,148 @@
+/*!
+*  Copyright (c) 2017 by Contributors
+* \file build_module.h
+* \brief Functions for compiling ops.
+*/
+#ifndef TVM_BUILD_MODULE_H_
+#define TVM_BUILD_MODULE_H_
+
+#include <string>
+#include "./tvm/c_dsl_api.h"
+#include "./tvm/runtime/packed_func.h"
+#include "./tvm/schedule_pass.h"
+#include "./tvm/lowered_func.h"
+
+namespace tvm {
+
+/*!
+* \brief Container for target device information.
+* Use target_llvm, target_cuda etc functions instead of constructing directly.
+*/
+struct Target {
+    /*! \brief The name of the target device */
+    std::string target_name;
+    /*! \brief The type of the target device */
+    DLDeviceType device_type;
+    /*! \brief The maximum threads that a schedule should use for this device */
+    int max_num_threads = 1;
+    /*! \brief The warp size that should be used by the LowerThreadAllreduce pass */
+    int thread_warp_size = 1;
+    /*! \brief Keys for this target */
+    std::unordered_set<std::string> keys;
+    /*! \brief Options for this target */
+    std::unordered_set<std::string> options;
```

Comment:
use std::vector<std::string> as later options might override previous ones, due de-duplication in string parsing